searchlib/src/tests/queryeval/wrappers/wrappers_test.cpp

```diff
@@ -0,0 +1,195 @@
+    search->initFullRange();
+    EXPECT_EQ(_data.unpackedDocId, 0u);
+    EXPECT_TRUE(!search->seek(1u));
+    EXPECT_EQ(search->getDocId(), 10u);
+    EXPECT_TRUE(search->seek(10));
+    search->unpack(10);
+    EXPECT_EQ(_data.unpackedDocId, 0u);
+    EXPECT_TRUE(!search->seek(15));
+    EXPECT_EQ(search->getDocId(), 20u);
+    EXPECT_TRUE(search->seek(20));
+    search->unpack(20);
+    EXPECT_EQ(_data.unpackedDocId, 0u);
+    EXPECT_TRUE(!search->seek(25));
+    EXPECT_TRUE(search->isAtEnd());
+
+    search.reset(nullptr);
+    EXPECT_EQ(_data.seekCnt, 6u);
+    EXPECT_EQ(_data.unpackCnt, 2u);
+    EXPECT_EQ(_data.dtorCnt, 2u);
+}
+
+TEST_F(WrapperTest, boolean_match_iterator_wrapper)
+{
+    verify_unwrapped();
+    { // with wrapper
+        TermFieldMatchData match;
+        TermFieldMatchDataArray tfmda;
+        tfmda.add(&match);
+        _data.unpackedDocId = 0;
+        auto to_wrap = std::make_unique<DummyItr>(_data, &match);
+        auto search = std::make_unique<BooleanMatchIteratorWrapper>(std::move(to_wrap), tfmda);
+        search->initFullRange();
+        EXPECT_EQ(_data.unpackedDocId, 0u);
+        EXPECT_TRUE(!search->seek(1u));
+        EXPECT_EQ(search->getDocId(), 10u);
+        EXPECT_TRUE(search->seek(10));
+        search->unpack(10);
+        EXPECT_EQ(_data.unpackedDocId, 0u);
+        EXPECT_TRUE(!search->seek(15));
+        EXPECT_EQ(search->getDocId(), 20u);
+        EXPECT_TRUE(search->seek(20));
+        search->unpack(20);
+        EXPECT_EQ(_data.unpackedDocId, 0u);
+        EXPECT_TRUE(!search->seek(25));
+        EXPECT_TRUE(search->isAtEnd());
+    }
+    EXPECT_EQ(_data.seekCnt, 6u);
+    EXPECT_EQ(_data.unpackCnt, 2u);
+    EXPECT_EQ(_data.dtorCnt, 2u);
+    { // with wrapper, without match data
+
+        auto to_wrap = std::make_unique<DummyItr>(_data, nullptr);
+        auto search = std::make_unique<BooleanMatchIteratorWrapper>(std::move(to_wrap), TermFieldMatchDataArray());
+        search->initFullRange();
+        EXPECT_TRUE(!search->seek(1u));
+        EXPECT_EQ(search->getDocId(), 10u);
+        EXPECT_TRUE(search->seek(10));
+        search->unpack(10);
+        EXPECT_TRUE(!search->seek(15));
+        EXPECT_EQ(search->getDocId(), 20u);
+        EXPECT_TRUE(search->seek(20));
+        search->unpack(20);
+        EXPECT_TRUE(!search->seek(25));
+        EXPECT_TRUE(search->isAtEnd());
+    }
+    EXPECT_EQ(_data.seekCnt, 9u);
+    EXPECT_EQ(_data.unpackCnt, 2u);
+    EXPECT_EQ(_data.dtorCnt, 3u);
+}
+
+class FilterWrapperVerifier : public search::test::SearchIteratorVerifier {
+public:
+    ~FilterWrapperVerifier() {}
+    SearchIterator::UP create(bool strict) const override {
+        auto search = std::make_unique<FilterWrapper>(1);
+        search->wrap(createIterator(getExpectedDocIds(), strict));
+        return search;
+    }
+};
+
+TEST(FilterWrapperTest, adheres_to_search_iterator_requirements)
+{
+    FilterWrapperVerifier verifier;
```

Comment:
How is the interaction between this app using gtest and SearchIteratorVerifier which uses legacy test macros? Consider using the functionality added in https://github.com/vespa-engine/vespa/pull/13728.

Reply:
enabled gtest version now.